lib/cisco_node_utils/itd_device_group.rb

```diff
@@ -0,0 +1,210 @@
+      create if instantiate
+    end
+
+    def self.itds
+      hash = {}
+      group_list = []
+      groups = config_get('itd_device_group',
+                          'all_itd_device_groups')
+      return hash if groups.nil?
+
+      groups.each do |group|
+        # The show cmd shows more than name if the probe is not confgured yet
+        # itd device-group abc frequency 10 timeout 5 retry-down-count 3 ...
+        # so filter it out to just get the name
+        group_list << group.split[0]
+      end
+
+      group_list.each do |id|
+        hash[id] = ItdDeviceGroup.new(id, false)
+      end
+      hash
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def create
+      Feature.itd_enable
+      config_set('itd_device_group', 'create', name: @name)
+    end
+
+    def destroy
+      config_set('itd_device_group', 'destroy', name: @name)
+    end
+
+    # Helper method to delete @set_args hash keys
+    def set_args_keys_default
+      @set_args = { name: @name }
+      @get_args = @set_args
+    end
+
+    def probe_params
+      params = config_get('itd_device_group', 'probe', @get_args)
+      params
+    end
+
+    # proble configuration is all done in a single line (like below)
+    # probe tcp port 32 frequency 10 timeout 5 retry-down-count 3 ...
+    # probe udp port 23 frequency 10 timeout 5 retry-down-count 3 ...
+    # probe icmp frequency 10 timeout 5 retry-down-count 3 retry-up-count 3
+    # probe dns host 8.8.8.8 frequency 10 timeout 5 retry-down-count 3 ...
+    # also the 'control enable' can be set if the type is tcp or udp only
+    # probe udp port 23 control enable frequency 10 timeout 5 ...
+    def probe
+      params = probe_params
+      hash = {}
+      hash[:probe_control] = default_probe_control
+      if params.nil?
+        hash[:probe_frequency] = default_probe_frequency
+        hash[:probe_timeout] = default_probe_timeout
+        hash[:probe_retry_down] = default_probe_retry_down
+        hash[:probe_retry_up] = default_probe_retry_up
+        hash[:probe_type] = default_probe_type
+        return hash
+      end
+      hash[:probe_frequency] = params[1].to_i
+      hash[:probe_timeout] = params[2].to_i
+      hash[:probe_retry_down] = params[3].to_i
+      hash[:probe_retry_up] = params[4].to_i
+
+      lparams = params[0].split
+      hash[:probe_type] = lparams[0]
+      case hash[:probe_type].to_sym
+      when :dns
+        hash[:probe_dns_host] = lparams[2]
+      when :tcp, :udp
+        hash[:probe_port] = lparams[2].to_i
+        hash[:probe_control] = true unless lparams[3].nil?
+      end
+      hash
+    end
+
+    def probe_control
+      hash = probe
+      hash[:probe_control]
+    end
+
+    def default_probe_control
+      config_get_default('itd_device_group', 'probe_control')
+    end
+
+    def probe_dns_host
+      hash = probe
```

Comment:
Do you need to handle the case where the `:probe_dns_host` key does not exist.. ie, `:probe_type` is `:tcp` or `:udp`?